src/commands/synthetics/run-test.ts

```diff
@@ -323,7 +325,11 @@ export class RunTestCommand extends Command {
     const testsToTrigger = suites
       .reduce((acc, suiteTests) => acc.concat(suiteTests), [])
       .map((test) => ({
-        config: {...this.config.global, ...test.config},
+        config: {
+          ...this.config.global,
+          ...test.config,
+          ...(this.config.locations?.length ? {locations: this.config.locations} : {}),
```

Comment:
Could you extract this object to a variable `configFromEnvironment`?
(If it could be created/populated outside of `this.config`/`resolveConfig` it'd be nice as well so that we know it is only used for overrides from env vars)